event_creation/submission/parsers/elemem_parsers.py

```diff
@@ -107,6 +108,444 @@ def event_stimulation(self, evdata):
+            final_object_recall_recording_stop=self.add_final_object_recall_recording_stop,
+            end_text=self.event_sess_end 
+            )
+	
+        self._add_type_to_modify_events(
+            final_object_recall_recording_start=self.modify_free_recall,
+            cued_recall_recording_start=self.modify_cued_rec,
+            final_store_recall_recording_start=self.modify_store_recall,
+            object_recall_recording_start=self.modify_rec_start,
+            stimming=self.modify_event_stimulation,
+            )
+
+    def _new_rec_event(self, recall, evdata):
+        word = recall[-1].strip().rstrip(".1") # some annotations have a .1 at the end?
+        new_event = self._empty_event
+        new_event.trial = self._trial
+        new_event.session = self._session
+        new_event.phase = 'practice' if self.practice else self.phase 
+        new_event.presX = self.presX
+        new_event.presZ = self.presZ
+        # new_event.rectime = int(round(float(recall[0])))
+        new_event.rectime = int(float(recall[0])) # old code did not round
+        new_event.mstime = evdata.mstime + new_event.rectime
+        new_event.msoffset = 20
+        new_event["item"] = word.upper()
+        new_event.itemno = int(recall[1])
+
+        return new_event
+
+    def _identify_intrusion(self, events, new_event):
+        words = events[events['type'] == 'WORD']
+        word_event = words[(words['item'] == new_event["item"])]
+        
+        if len(word_event) > 1:
+            raise Exception("Repeat items not supported or expected. Please check your data.")
+
+        elif len(word_event) == 0: #ELI
+            new_event.intrusion  = -1
+
+        elif word_event["trial"][0] == self._trial:
+            new_event.intrusion = 0
+            new_event.serialpos = words[words['item'] == new_event["item"]]['serialpos'][0]
+            new_event.store  = words[words['item'] == new_event["item"]]['store'][0]
+            new_event.storeX = words[words['item'] == new_event["item"]]['storeX'][0]
+            new_event.storeZ = words[words['item'] == new_event["item"]]['storeZ'][0]
+
+        elif word_event["trial"][0] >= 0: # PLI
+            new_event.intrusion = self._trial - word_event["trial"][0]
+            new_event.serialpos = words[words['item'] == new_event["item"]]['serialpos'][0]
+            new_event.store  = words[words['item'] == new_event["item"]]['store'][0]
+            new_event.storeX = words[words['item'] == new_event["item"]]['storeX'][0]
+            new_event.storeZ = words[words['item'] == new_event["item"]]['storeZ'][0]
+        else:
+            raise Exception("Processing error, word event was not presented during experimental trial")
+
+        return new_event
+
+    ####################
+    # Functions to add new events from a single line in the log
+    ####################
+
+    # keep track of stimtag information for later
+    def event_stimtag(self, evdata):
```

Comment:
This should go in another field in stim params, not a new "stimtag" field.

Reply:
This function is purely for saving the current stim frequency (3 or 8Hz) as a local variable, so that I can use that information later on when populating the stim_params!

Reply:
This should be implemented as part of the BaseElememParser since it's abstractly an elemem configuration / system event. It's not specific to this experiment. Warrants discussion with @rcolyer since he added this event to the network protocol after I wrote the original parser.

Reply:
I decided to leave it in so that I can access what burst frequency the stimulation was at. But "stimtag" is not added to the processed dataframe.